### PR TITLE
Implemented Store model w/ migrations

### PR DESCRIPTION
**Pull request by kjcioffi**
* Added store model to models.py
* Made migration file
* Created a test suite with edge cases for the model.
* Made Store and User factories to support the Store Model test suite and future test suites.
* Added new dev dependencies to support the project.

### CODE REVIEW

File: store/tests/models/test_store_model.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+    def test_store_doesnt_delete_cascade_user(self):
+        self.store.delete()
+        self.assertTrue(User.objects.filter(pk=self.store.owner.pk).exists())
+
+    def test_city_within_25_chars(self):
+        self.assertLessEqual(
+            len(self.store.city),
+            self.city_char_length,
+            f"Store name field must be {self.city_char_length} characters or less.",
```
Comment (ryaustin) [inline store/tests/models/test_store_model.py:56]: check for consistency here. Should probably be 'City' not 'Store'

File: store/tests/models/test_store_model.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+        )
+        self.store.city = city
+
+        with self.assertRaises(ValidationError):
+            self.store.clean_fields()
+
+        self.assertEqual(
+            len(self.store.city),
+            self.city_char_length + 1,
+            f"Store name field must be {self.city_char_length} characters or less.",
```
Comment (ryaustin) [inline store/tests/models/test_store_model.py:72]: again here - check for error message consistency.

File: store/tests/models/test_store_model.py [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+        )
+
+    def test_state_within_25_chars(self):
+        self.assertLessEqual(
+            len(self.store.state),
+            self.state_char_length,
+            f"Store name field must be {self.state_char_length} characters or less.",
```
Comment (ryaustin) [inline store/tests/models/test_store_model.py:79]: 👆'State'

File: store/tests/models/test_store_model.py [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+        self.store.state = state
+
+        with self.assertRaises(ValidationError):
+            self.store.clean_fields()
+
+        self.assertEqual(
+            len(self.store.state),
+            self.state_char_length + 1,
+            f"Store name field must be {self.state_char_length} characters or less.",
+        )
```
Comment (ryaustin) [inline store/tests/models/test_store_model.py:96]: 👆'State'

**Review by ryaustin — APPROVED** [review]
Looking good, just some consistency in error messages but small changes.
- Consider adding some edge cases if important such as strings exactly at max length and if empty strings are allowed.

**Comment by kjcioffi** [comment]
Thank you 🙏, @ryaustin. I was initially a bit confused by what you meant, I appreciate your attention to the finer details.

**Review by ryaustin — APPROVED** [review]
Looking good!

**Comment by kjcioffi** [comment]
Edge cases were added as recommended and failure messages were re-phrased for better clarity.
